projects/plugins/crm/src/event-manager/class-events-manager.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Events_Manager class.
+ *
+ * @package Automattic\Jetpack\CRM\Event_Manager
+ */
+
+namespace Automattic\Jetpack\CRM\Event_Manager;
+
+/**
+ * Events_Manager class.
+ *
+ * @since $$next-version$$
+ */
+class Events_Manager {
+
+	/** @var null The Events_Manager instance. */
+	private static $instance = null;
+
+	/** @var bool Whether or not the event manager is active. */
+	private $active = true;
+
+	/**
+	 * Get the singleton instance of this class.
+	 *
+	 * @return Events_Manager
+	 */
+	public static function get_instance(): Events_Manager {
+		if ( ! self::$instance ) {
+			self::$instance = new Events_Manager();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Set the Events_Manager instance for testing purposes.
+	 *
+	 * @param Events_Manager $instance The Events_Manager instance.
+	 * @return void
+	 */
+	public static function set_instance( Events_Manager $instance ) {
+		self::$instance = $instance;
+	}
+
+	/**
+	 * Set whether or not the event manager is active.
+	 *
+	 * @param bool $active Whether or not the event manager is active.
+	 * @return void
+	 */
+	public function set_active( bool $active ) {
+		$this->active = $active;
+	}
+
+	/**
+	 * Get whether or not the event manager is active.
+	 *
+	 * @return bool
+	 */
+	public function is_active(): bool {
```

Comment:
I don't see this being used anywhere in the system. IMHO, we should remove this and address in a separate PR, or use this flag where it is needed and add tests.

Reply:
True. I started to add it after talking about the possibility of switching off the notifications and in consequence the automation (i.e. on migrations). But I didn't continue with it. And yes, better if we address it in a separate PR. Thanks.

Reply:
Fixed here: https://github.com/Automattic/jetpack/pull/32139/commits/b9aa4866c2006b72e52da6e2f4eafee33b697709